internal/services/monitor/monitor_scheduled_query_rules_alert_v2_resource.go

```diff
@@ -0,0 +1,895 @@
+	Threshold           float64                                         `tfschema:"threshold"`
+	TimeAggregation     scheduledqueryrules.TimeAggregation             `tfschema:"time_aggregation"`
+}
+
+type ScheduledQueryRulesAlertV2DimensionModel struct {
+	Name     string                                `tfschema:"name"`
+	Operator scheduledqueryrules.DimensionOperator `tfschema:"operator"`
+	Values   []string                              `tfschema:"values"`
+}
+
+type ScheduledQueryRulesAlertV2FailingPeriodsModel struct {
+	MinFailingPeriodsToAlert  int64 `tfschema:"min_failing_periods_to_alert"`
+	NumberOfEvaluationPeriods int64 `tfschema:"number_of_evaluation_periods"`
+}
+
+type ScheduledQueryRulesAlertV2Resource struct{}
+
+var _ sdk.ResourceWithUpdate = ScheduledQueryRulesAlertV2Resource{}
+
+func (r ScheduledQueryRulesAlertV2Resource) ResourceType() string {
+	return "azurerm_monitor_scheduled_query_rules_alert_v2"
+}
+
+func (r ScheduledQueryRulesAlertV2Resource) ModelObject() interface{} {
+	return &ScheduledQueryRulesAlertV2Model{}
+}
+
+func (r ScheduledQueryRulesAlertV2Resource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return scheduledqueryrules.ValidateScheduledQueryRuleID
+}
+
+func (r ScheduledQueryRulesAlertV2Resource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"criteria": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Resource{
+
+				Schema: map[string]*pluginsdk.Schema{
+					"query": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"operator": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(scheduledqueryrules.ConditionOperatorEquals),
+							string(scheduledqueryrules.ConditionOperatorGreaterThan),
+							string(scheduledqueryrules.ConditionOperatorGreaterThanOrEqual),
+							string(scheduledqueryrules.ConditionOperatorLessThan),
+							string(scheduledqueryrules.ConditionOperatorLessThanOrEqual),
+						}, false),
+					},
+
+					"time_aggregation": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(scheduledqueryrules.TimeAggregationCount),
+							string(scheduledqueryrules.TimeAggregationAverage),
+							string(scheduledqueryrules.TimeAggregationMinimum),
+							string(scheduledqueryrules.TimeAggregationMaximum),
+							string(scheduledqueryrules.TimeAggregationTotal),
+						}, false),
+					},
+
+					"threshold": {
```

Comment:
is this a persntage?
```suggestion
					"threshold_percentage": {
```

Reply:
This threshold can accpect float values.

Reply:
sure, could we add some validation thou?

Reply:
Makes sense to me!